pyfar/dsp/dsp.py

```diff
@@ -2669,3 +2669,171 @@ def normalize(signal, reference_method='max', domain='auto',
+        `signal_1` and its ``cshape`` must be broadcastable to that of
+        `signal_1`.
+    mode : str, optional
+
+        ``'full'``
+            Computes the full correlation by zero padding the input signals
+            to a length of ``signal_1.n_samples + signal_2.n_samples - 1``
+            before applying the Fourier transform (see equations above).
+        ``'cyclic'``
+            Computes the cyclic correlation, which uses the input signals as
+            they are. In this case `signal_1` and `signal_2` must have the
+            same number of samples (length).
+
+        The default is ``'full'``.
+
+    Returns
+    -------
+    correlation : TimeData
+        The correlation :math:`c[l]` is contained in ``correlation.time`` and
+        the lags, i.e., the delays applied to `signal_2` (see equations above)
+        are contained in ``correlation.times``. The cshape of `correlation`
+        matches the cshape to which `signal_1` and `signal_2` were broadcasted.
+        The time lags are given in seconds and can be converted to samples by
+        multiplication with ``signal_1.sampling_rate``. In this case, they are
+        in the interval ``[-signal_2.n_samples + 1, signal_1.n_samples - 1]``
+        if the full correlation was computed. In case of the cyclic
+        correlation, they are in the interval
+        ``[-(signal_1.n_samples // 2) + 1, signal_1.n_samples // 2]``
+        if the signals have an even number of samples, and in the interval
+        ``[-(signal_1.n_samples // 2), signal_1.n_samples // 2]``
+        if the signals have an odd number of samples.
+
+    Examples
+    --------
+
+    Linear and cyclic auto-correlation of a perfect sequence. Perfect sequences
+    have unit auto-correlation for :math:`l=0` and zero auto-correlation
+    otherwise
+
+    .. plot::
+
+        >>> import pyfar as pf
+        >>>
+        >>> signal = pf.signals.linear_perfect_sweep(2**8)
+        >>>
+        >>> for mode in ['full', 'cyclic']:
+        >>>     cor = pf.dsp.correlate(signal, signal, mode)
+        >>>     ax = pf.plot.time(cor / pf.dsp.energy(signal), label=mode)
+        >>>
+        >>> ax.set_xlabel('time lags in seconds')
+        >>> ax.set_ylabel('auto correlation')
+        >>> ax.legend()
+
+    Compute the lags (delay) that are required to maximize the
+    cross-correlation between a one and multi-dimensional signal
+
+    .. plot::
+
+        >>> import pyfar as pf
+        >>> import numpy as np
+        >>>
+        >>> # one-dimensional signal: impulse with zero delay
+        >>> signal_1 = pf.signals.impulse(5, 0)
+        >>>
+        >>> # multi-dimension signal with cshape = (2, 2):
+        >>> # impulses with non-zero delays
+        >>> delays = np.array([[0, 1], [2, 3]], dtype=int)
+        >>> signal_2 = pf.signals.impulse(5, delays)
+        >>>
+        >>> cor = pf.dsp.correlate(signal_1, signal_2, 'full')
```

Comment:
why not to plot the correlation vs lags in samples here? and then mark the argmax with a vertical line

Reply:
Did that